libnetwork/netutils/utils_linux_test.go

```diff
@@ -186,21 +190,44 @@ func TestNetworkRange(t *testing.T) {
 
 // Test veth name generation "veth"+rand (e.g.veth0f60e2c)
 func TestGenerateRandomName(t *testing.T) {
-	name1, err := GenerateRandomName("veth", 7)
-	if err != nil {
-		t.Fatal(err)
-	}
-	// veth plus generated append equals a len of 11
-	if len(name1) != 11 {
-		t.Fatalf("Expected 11 characters, instead received %d characters", len(name1))
-	}
-	name2, err := GenerateRandomName("veth", 7)
-	if err != nil {
-		t.Fatal(err)
-	}
-	// Fail if the random generated names equal one another
-	if name1 == name2 {
-		t.Fatalf("Expected differing values but received %s and %s", name1, name2)
+	const vethPrefix = "veth"
+	const vethLen = len(vethPrefix) + 7
+
+	testCases := []struct {
+		prefix string
+		length int
+		error  bool
+	}{
+		{vethPrefix, -1, true},
+		{vethPrefix, 0, true},
+		{vethPrefix, len(vethPrefix) - 1, true},
+		{vethPrefix, len(vethPrefix), true},
+		{vethPrefix, len(vethPrefix) + 1, false},
+		{vethPrefix, 255, false},
+	}
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("prefix=%s,length=%d", tc.prefix, tc.length), func(t *testing.T) {
```

Comment:
Generally, slash (`/`) characters are the preferred choice to delimit the parts of a subtest name [due to the behaviour of the `go test -run` flag.](https://pkg.go.dev/cmd/go#hdr-Testing_flags:~:text=%2Drun%20regexp%0A%20%20%20%20Run,those%20sub%2Dtests.) E.g.:
```console
$ go test -run TestGenerateRandomName//length=0 ./libnetwork/netutils
```
will focus on the subtests where length=0, irrespective of the prefix.

```suggestion
		t.Run(fmt.Sprintf("prefix=%s/length=%d", tc.prefix, tc.length), func(t *testing.T) {
```

(This is one of those obscure pieces of tribal knowledge tucked away in the documentation for the `go` command instead of somewhere actually discoverable, like the `"testing"` package docs)

Reply:
I did not know that you could leave path segments empty -- neat!